frontend/src/metabase/query_builder/components/QuestionActions.unit.spec.tsx

```diff
@@ -46,22 +64,55 @@ function setup({ card }: { card: Card }) {
   );
 }
 
-describe("Question Actions | Icons", () => {
-  ["structured", "native"].forEach(queryType => {
-    iconList.forEach(({ label, tooltipText }) => {
-      it(`should display the "${label}" icon with the "${tooltipText}" tooltip for ${queryType} questions`, async () => {
-        setup({
-          card:
-            queryType === "structured"
-              ? TEST_STRUCTURED_CARD
-              : TEST_NATIVE_CARD,
-        });
+describe("QuestionActions", () => {
+  it.each(ICON_CASES)(
+    `should display the "$label" icon with the "$tooltipText" tooltip for $card.name questions`,
+    async ({ label, tooltipText, card }) => {
+      setup({ card });
+
+      await userEvent.hover(screen.getByRole("button", { name: label }));
+      const tooltip = screen.getByRole("tooltip", { name: tooltipText });
+      expect(tooltip).toHaveAttribute("data-placement", "top");
+      expect(tooltip).toHaveTextContent(tooltipText);
+    },
+  );
 
-        await userEvent.hover(screen.getByRole("button", { name: label }));
-        const tooltip = screen.getByRole("tooltip", { name: tooltipText });
-        expect(tooltip).toHaveAttribute("data-placement", "top");
-        expect(tooltip).toHaveTextContent(tooltipText);
-      });
+  it("should allow to edit the model only with write permissions", () => {
```

Comment:
This test does not assert that it happens "only" with write permissions

```suggestion
  it("should allow to edit the model with write permissions", () => {
```